test/integration/tas/tas_job_test.go

```diff
@@ -0,0 +1,602 @@
+				gomega.Expect(k8sClient.Create(ctx, tasFlavor)).Should(gomega.Succeed())
+
+				clusterQueue = testing.MakeClusterQueue("cluster-queue").
+					ResourceGroup(*testing.MakeFlavorQuotas(tasFlavor.Name).Resource(corev1.ResourceCPU, "5").Obj()).
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+
+				localQueue = testing.MakeLocalQueue("local-queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+				gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+			})
+
+			ginkgo.AfterEach(func() {
+				gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+				gomega.Expect(util.DeleteWorkloadsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+				gomega.Expect(util.DeleteObject(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+				gomega.Expect(util.DeleteObject(ctx, k8sClient, topology)).Should(gomega.Succeed())
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, tasFlavor, true)
+			})
+
+			ginkgo.It("should admit workload when nodes become available", func() {
+				var (
+					nodes []corev1.Node
+					wl1   *kueue.Workload
+				)
+
+				ginkgo.By("creating a workload which requires rack, but does not fit in any", func() {
+					wl1 = testing.MakeWorkload("wl1", ns.Name).
+						Queue(localQueue.Name).Request(corev1.ResourceCPU, "1").Obj()
+					wl1.Spec.PodSets[0].TopologyRequest = &kueue.PodSetTopologyRequest{
+						Required: ptr.To(tasRackLabel),
+					}
+					gomega.Expect(k8sClient.Create(ctx, wl1)).Should(gomega.Succeed())
+				})
+
+				ginkgo.By("verify the workload is inadmissible", func() {
+					util.ExpectPendingWorkloadsMetric(clusterQueue, 0, 1)
+				})
+
+				ginkgo.By("Create nodes to allow scheduling", func() {
+					nodes = []corev1.Node{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "b1-r1",
+								Labels: map[string]string{
+									tasBlockLabel: "b1",
+									tasRackLabel:  "r1",
+								},
+							},
+							Status: corev1.NodeStatus{
+								Allocatable: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("1"),
+									corev1.ResourceMemory: resource.MustParse("1Gi"),
+								},
+							},
+						},
+					}
+					for _, node := range nodes {
+						gomega.Expect(k8sClient.Create(ctx, &node)).Should(gomega.Succeed())
+						gomega.Expect(k8sClient.Status().Update(ctx, &node)).Should(gomega.Succeed())
+					}
+				})
+
+				ginkgo.By("verify the workload is admitted", func() {
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl1)
+					util.ExpectReservingActiveWorkloadsMetric(clusterQueue, 1)
+					util.ExpectPendingWorkloadsMetric(clusterQueue, 0, 0)
+				})
+
+				ginkgo.By("Delete the node for cleanup", func() {
+					for _, node := range nodes {
+						gomega.Expect(k8sClient.Delete(ctx, &node)).Should(gomega.Succeed())
```

Comment:
Introducing "util.ExpectObjectToBeDeleted".

Reply:
Done in the follow up https://github.com/kubernetes-sigs/kueue/pull/3291